test-data/unit/check-custom-plugin.test

```diff
@@ -280,3 +280,30 @@ reveal_type(FullyQualifiedTestNamedTuple('')._asdict()) # E: Revealed type is 'b
 [[mypy]
 plugins=<ROOT>/test-data/unit/plugins/fully_qualified_test_hook.py
 [builtins fixtures/classmethod.pyi]
+
+[case testDynamicClassPlugin]
+# flags: --config-file tmp/mypy.ini
+from mod import declarative_base, Column, Instr
+
+Base = declarative_base()
```

Comment:
Also test a few different assignment statements and ensure that the plugin doesn't get run them. Maybe things like `A = B = ...` and `C = <other_function>()`. If the plugin doesn't get run, the lvalue shouldn't be a valid base class.